core/domain/question_domain.py

```diff
@@ -220,14 +251,20 @@ def update_state_from_model(
                     state data.
             current_state_schema_version: int. The current state
                 schema version.
+            question_model: QuestionModel. The question model loaded from the
+                datastore.
         """
         versioned_question_state['state_schema_version'] = (
             current_state_schema_version + 1)
 
         conversion_fn = getattr(cls, '_convert_state_v%s_dict_to_v%s_dict' % (
             current_state_schema_version, current_state_schema_version + 1))
-        versioned_question_state['state'] = conversion_fn(
-            versioned_question_state['state'])
+        if current_state_schema_version == 29:
+            versioned_question_state['state'] = conversion_fn(
+                versioned_question_state['state'], question_model)
```

Comment:
Maybe just pass the linked_skill_id instead of the entire question?

Reply:
Done.